ostelco-core/Models/StageDecider.swift

```diff
@@ -131,50 +111,60 @@ public struct StageDecider {
             remove(.eSimInstructions)
         }
         
-        if let profile = region.getSimProfile(), profile.status == .installed {
+        if let profile = region.getGraphQLModel().getSimProfile(), profile.status == .installed {
             remove(.eSimOnboarding)
             remove(.eSimInstructions)
-            
-            if !localContext.hasSeenESimOnboarding {
-                remove(.awesome)
-            }
-        }
-        
-        if localContext.hasSeenAwesome {
-            remove(.awesome)
         }
         return stages[0]
     }
     
-    // swiftlint:disable:next cyclomatic_complexity
     public func compute(context: Context?, localContext: LocalContext) -> Stage {
         // Error Stages
         if localContext.serverIsUnreachable {
             return .ohNo(.serverUnreachable)
         }
         
-        if let problem = localContext.locationProblem {
-            return .locationProblem(problem)
+        if context?.customer != nil {
+            // This is a clue the user is an existing user, don't need to show legal stuff
+            return .home
         }
         
-        // Initial Stages
-        guard let context = context else {
-            return preLoggedInStage(localContext)
+        var stages: [Stage] = [.loginCarousel, .legalStuff, .nicknameEntry, .home]
+        
+        func remove(_ stage: Stage) {
+            if let index = stages.firstIndex(of: stage) {
+                stages.remove(at: index)
+            }
         }
         
+        if localContext.hasFirebaseToken {
+            remove(.loginCarousel)
+        }
+        if localContext.hasAgreedToTerms {
+            remove(.legalStuff)
+        }
+        return stages[0]
+    }
+    
+    // swiftlint:disable:next cyclomatic_complexity
```

Comment:
What causes high cyclomatic complexity here?  There should be a comment explaining why the warning
is disabled.

Reply:
The method just below it has quite a few branching statements, that's what cyclomatic complexity is. The warning is disabled because this is the one place where we're allowing it since we know why.